packages/starknet-snap/src/utils/starknetUtils.ts

```diff
@@ -797,11 +824,15 @@ export async function createAccount({
   }
 
   if (waitMode) {
-    await waitForTransaction(
+    // In `waitMode`, we need to ensure that the account deployment is fully confirmed
+    // before proceeding with other transactions.
+    // Instead of waiting for a specific transaction to complete (`waitForTransaction`),
+    // we use `waitForBlock` to wait until the `Pending` block get's accepted.
+    await waitForBlock(
```

Comment:
i think it is better to stay with waitForTransaction
if you double check the function inside the starknet.js, they are using waitForTransaction internally for some deploy contract method as well

what is the actual issue with waitForTransaction?